app/controllers/storytime/posts_controller.rb

```diff
@@ -10,11 +13,17 @@ def index
     def show
       params[:id] = params[:id].split("/").last
 
-      @post = if params[:preview]
-        Post.find_preview(params[:id])
-      else
-        Post.published.friendly.find(params[:id])
+      @post = begin
+        if params[:preview]
+          Post.find_preview(params[:id])
+        else
+          Post.published.friendly.find(params[:id])
+        end
+      rescue ActiveRecord::RecordNotFound
+        nil
       end
+      
+      return super if @post.nil?
```

Comment:
Valid. The only thing with that is that we're using FriendlyID, so params[:id] could be and ID or slug.